libs/domain/picking/src/components/picking-product-card/picking-product-card.component.spec.ts

```diff
@@ -0,0 +1,267 @@
+          'oryx-button button'
+        ) as HTMLButtonElement;
+        button.click();
+      });
+
+      it('should dispatch edit event', () => {
+        expect(spy).toHaveBeenCalled();
+      });
+    });
+
+    describe('when a product is not provided', () => {
+      beforeEach(async () => {
+        element = await fixture(
+          html`
+            <oryx-picking-product-card
+              status="${ItemsFilters.NotPicked}"
+            ></oryx-picking-product-card>
+          `
+        );
+      });
+
+      it('should not render form quantity input', async () => {
+        expect(
+          element.shadowRoot?.querySelector(
+            'oryx-cart-quantity-input'
+          ) as QuantityInputComponent
+        ).to.not.exist;
+      });
+    });
+  });
+
+  describe('when a product provided with status Picked', () => {
+    beforeEach(async () => {
+      element = await fixture(
+        html`
+          <oryx-picking-product-card
+            .productItem="${productItem}"
+            status="${ItemsFilters.Picked}"
+          ></oryx-picking-product-card>
+        `
+      );
+    });
+
+    it('should not render quantity input', () => {
+      expect(
+        element.shadowRoot?.querySelector(
+          'oryx-cart-quantity-input'
+        ) as QuantityInputComponent
+      ).to.not.exist;
+    });
+
+    it('should render summary info', () => {
+      expect(element.shadowRoot?.querySelector('.summary-info')).to.exist;
+    });
+
+    it('should render edit button', () => {
+      expect(
+        element.shadowRoot
+          ?.querySelector('oryx-button button')
+          ?.textContent?.trim()
+      ).toBe('Edit items');
+    });
+  });
+
+  describe('when a product provided with status Not Found', () => {
+    beforeEach(async () => {
+      element = await fixture(
+        html`
+          <oryx-picking-product-card
+            .productItem="${productItem}"
+            status="${ItemsFilters.NotFound}"
+          ></oryx-picking-product-card>
+        `
+      );
+    });
+
+    it('should render oryx-image with the image-fade class', () => {
+      expect(
+        element.renderRoot
+          .querySelector('oryx-image')
+          ?.classList.value.includes('image-fade')
+      ).toBe(true);
+
+      expect(
+        (element.shadowRoot?.querySelector('oryx-image') as ImageComponent).src
+      ).toBe(productItem.product.image);
+    });
+
+    it('should not render quantity input', () => {
+      expect(
+        element.shadowRoot?.querySelector(
+          'oryx-cart-quantity-input'
+        ) as QuantityInputComponent
+      ).to.not.exist;
+    });
+
+    it('should render summary info', () => {
+      expect(element.shadowRoot?.querySelector('.summary-info')).to.exist;
+    });
+
+    it('should render edit button', () => {
+      expect(
+        element.shadowRoot
+          ?.querySelector('oryx-button button')
+          ?.textContent?.trim()
+      ).toBe('Edit items');
```

Comment:
Do we need to double check presence of this button for Picked and Not found states, because we can just check whether we have `.summary-info` or not. This button is a part of summary info, and there is no conditions for its rendering

And I don't see tests that cover `renderEditStatus` method

Reply:
It's not really necessary, but I would say it doesn't hurt either.
The renderEditStatus it's being test wherever the Status is Picked or NotFound